### PR TITLE
Change `exec` from alias to function to handle arbitrary args

### LINKED ISSUE #18561: ssh command can't spawn a proxy command via "exec" from a pwsh 7.3.0 session
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

When using an `ssh` command with a proxy host (`-J`) parameter on a Mac, it tries to relaunch the command using `exec`. Before 7.3.0 this worked through a workaround `exec` file that was created in `/usr/local/microsoft/powershell/7`:
```powershell
#!/usr/bin/env pwsh
param($command)

& $command @args
```

Now, after 7.3.0 upgrade, with or without the "exec" file, the command returns an error whenever attempting to spawn a proxy command.

### Expected behavior

```console
(bash)$ ssh 10.1.0.12 -J proxyhost.local -v
OpenSSH_8.6p1, LibreSSL 3.3.6
debug1: Setting implicit ProxyCommand from ProxyJump: ssh -v -W '[%h]:%p' proxyhost.local
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Executing proxy command: exec ssh -v -W '[10.1.0.12]:22' proxyhost.local
debug1: identity file /Users/user/.ssh/id_rsa type 0
debug1: identity file /Users/user/.ssh/id_rsa-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.6
OpenSSH_8.6p1, LibreSSL 3.3.6
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Connecting to proxyhost.local port 22.
debug1: Connection established.
```

### Actual behavior

```console
PS> ssh 10.1.0.12 -J proxyhost.local -v
OpenSSH_8.6p1, LibreSSL 3.3.6
debug1: Setting implicit ProxyCommand from ProxyJump: /usr/bin/ssh -v -W '[%h]:%p' proxyhost.local
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Executing proxy command: exec /usr/bin/ssh -v -W '[10.1.0.12]:22' proxyhost.local
debug1: identity file /Users/user/.ssh/id_rsa type 0
debug1: identity file /Users/user/.ssh/id_rsa-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.6
debug1: kex_exchange_identification: banner line 0: \033[31;1mSwitch-Process: \033[31;1mA positional parameter cannot be found that accepts argument '/usr/bin/ssh'.\033[0m
kex_exchange_identification: Connection closed by remote host
Connection closed by UNKNOWN port 65535
```

### Error details

```console
$null
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Darwin 21.6.0 Darwin Kernel Version 21.6.0: Thu Sep 29 20:13:56 PDT 2022; root:xnu-8020.240.7~1/RELEASE_ARM64_T6000
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

image

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Previously `exec` was an alias to `switch-process`.  The problem is if you use `exec` with args that contain something like `-w` which PowerShell tries to bind to `-WithCommand` (parameter for `Switch-Process`) then the other args try to get bound to positional parameters which do not exist.  So something like `exec /bin/echo 1 -w 2` fails whereas it should simply replace the current `pwsh` with `echo` and emit "1 -w 2".

The fix is to change from an alias to a function so that all the args get passed to `Switch-Parameter` by using `-WithCommand` explicitly so the remaining args get handled correctly.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/18561

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/9444
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by daxian-dbw — APPROVED** [review]
LGTM

File: test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1 [hunk 1]
```diff
@@ -60,6 +60,11 @@ Describe 'Switch-Process tests for Unix' -Tags 'CI' {
         $env = pwsh -noprofile -outputformat text -command { $env:TEST_FOO='my test = value'; Switch-Process bash -c 'echo $TEST_FOO' }
         $env | Should -BeExactly 'my test = value'
     }
+
+    It 'The command can include a -w paramter' {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1:64]:
```suggestion
    It 'The command can include a -w parameter' {
```

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 2]
```diff
@@ -4078,6 +4078,13 @@ internal static string GetClearHostFunctionText()
             }
         }
 
+        internal static string GetExecFunctionText()
+        {
+            return @"
+Switch-Process -WithCommand $args
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/InitialSessionState.cs:4084]:
```suggestion
Switch-Process -WithCommand $args
```

**Review by daxian-dbw — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 3]
```diff
@@ -4078,6 +4078,13 @@ internal static string GetClearHostFunctionText()
...
+";
+        }
+
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/InitialSessionState.cs:4087]: Should it be in `#if UNIX`?

File: test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1 [hunk 4]
```diff
@@ -15,10 +15,10 @@ Describe 'Switch-Process tests for Unix' -Tags 'CI' {
         $global:PSDefaultParameterValues = $originalDefaultParameterValues
     }
 
-    It 'Exec alias should map to Switch-Process' {
+    It 'Exec function should map to Switch-Process' {
         $alias = Get-Command exec
```
Comment (kilasuit) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1:19]: for cleanliness any variables that were `$alias` should be `$function` as not to accidentality confuse future readers of the codebase

**Review by kilasuit — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1 (hunk 4 above)
Comment (kilasuit) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1:19]: This is just one example & I would expect that all other usage of `$alias` should be changed without me needing to highlight it as it's only a few additional lines

**Review by iSazonov — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1 (hunk 4 above)
Comment (daxian-dbw) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1]: Addressed, thanks!

Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Core/Exec.Tests.ps1]: good point, will update

**Pull request merged**